lib/nrf_modem_lib/Kconfig

```diff
@@ -49,24 +57,16 @@ choice NRF_MODEM_LIB_TRACE_MEDIUM
 
 config NRF_MODEM_LIB_TRACE_MEDIUM_UART
 	bool "Send modem trace over UARTE1"
-	# Modem tracing over UART use the UARTE1 as dedicated peripheral.
-	# This enable UARTE1 peripheral and includes nrfx UARTE driver.
-	select NRFX_UARTE1
```

Comment:
Why have these been moved to the `NRF_MODEM_LIB_TRACE_ENABLED` option?
This way enabling traces will select both `NRFX_UARTE1` and `USE_SEGGER_RTT` regardless of the selected trace medium.

Reply:
These had to be moved to aid the compilation of the unit tests. I have verified (by comparing the .map files) that the Segger APIs dont get compiled in when UART medium is selected and vice versa.

More explanation: I wrote the unit test in a way that the trace transport medium can be controlled by the test case at run time. I used the IS_ENABLED() macro for this purpose in modem_trace.c instead of the regular #ifdef or #if. This is the recommended way of doing this kind of unit testing. (See https://github.com/nrfconnect/sdk-nrf/blob/c3e71088c1210cc670f1c9a86f8d70fb6b053b20/tests/unity/example_test/src/example_test.h#L12 and also https://www.kernel.org/doc/html/latest/process/coding-style.html#conditional-compilation).

The only drawback is the way we write the KConfig. But the benefit is cleaner code (in .c files) and testability.

Reply:
> I wrote the unit test in a way that the trace transport medium can be controlled by the test case at run time. 

I am not convinced we should change these to accommodate the specific way these unit tests are written. You are introducing a dependency on this setting. It may be that one board does not have RTT, and then `NRF_MODEM_LIB_TRACE_ENABLED` can't `select USE_SEGGER_RTT` it. What happens then?

I suggest you create two unit tests suites, one that compiles with the UART backend and one with RTT backend. That will also be easier to maintain when we eventually split the backends.

Reply:
Alright I will give that a shot.

Reply:
@lemrey  I have split of the unit tests as you suggested and made the `nrf_modem_lib/Kconfig` look similar to that in ncs/main. But I will have to delete the uart test because of the way I mock/hack the device tree in the test's `CMakelists.txt` is not accepted. But I think you can still have a look at the `nrf_modem_lib/Kconfig` and see if that is satisfactory.